src/base/BaseSwapRouter.sol

```diff
@@ -129,9 +128,15 @@ abstract contract BaseSwapRouter is SafeCallback {
                     permitPayload.permit,
                     permitPayload.signature
                 );
-                outputCurrency.take(poolManager, data.to, outputAmount, false);
             } else {
+                // Handle regular ERC20...
                 inputCurrency.settle(poolManager, data.payer, inputAmount, false);
+            }
+
+            // For ERC6909 output, mint directly to recipient
+            if (data.outputIs6909) {
+                poolManager.mint(data.to, outputCurrency.toId(), outputAmount);
+            } else {
                 outputCurrency.take(poolManager, data.to, outputAmount, false);
             }
```

Comment:
i think because were using CurrencySettler you can use instead of if/else

`outputCurrency.take(poolManager, data.to, outputAmount, data.outputIs6909);`